ui/app/components/clients/activity.ts

```diff
@@ -65,18 +64,18 @@ export default class ClientsActivityComponent extends Component<Args> {
       return activity.byMonth;
     }
     const namespaceData = activity.byMonth
-      .map((m) => m.namespaces_by_key[namespace as keyof typeof m.namespaces_by_key])
+      .map((m) => m.namespaces_by_key[namespace])
       .filter((d) => d !== undefined);
 
     if (!mountPath) {
-      return namespaceData.length === 0 ? undefined : namespaceData;
+      return !namespaceData ? [] : namespaceData;
     }
 
-    const mountData = mountPath
-      ? namespaceData.map((namespace) => namespace?.mounts_by_key[mountPath]).filter((d) => d !== undefined)
-      : namespaceData;
+    const mountData = namespaceData
+      .map((namespace) => namespace?.mounts_by_key[mountPath])
+      .filter((d) => d !== undefined);
 
-    return mountData.length === 0 ? undefined : mountData;
+    return !mountData ? [] : mountData;
```

Comment:
Again, as long as `namespaceData` is an array, `mountData` will always be an array and return truthy